frontend/src/app/modules/main/create-edit-playlist/create-edit-playlist/create-edit-playlist.component.ts

```diff
@@ -0,0 +1,209 @@
+import { Router, ActivatedRoute } from '@angular/router';
+import { EditedPlaylist } from 'src/app/models/playlist/editedPlaylist';
+import { User } from 'src/app/models/user/user';
+
+@Component({
+  selector: 'app-create-edit-playlist',
+  templateUrl: './create-edit-playlist.component.html',
+  styleUrls: ['./create-edit-playlist.component.sass']
+})
+export class CreateEditPlaylistComponent implements OnInit, OnDestroy {
+  public isModalShown = false;
+  public foundSongs: Array<Song> = new Array<Song>();
+  public playlistSongs: Array<Song> = new Array<Song>();
+  public previousPlaylistData: EditedPlaylist;
+  public playlist = {} as Playlist;
+
+  private _id: number | undefined;
+
+  private _searchTerms = new Subject<string>();
+  private _unsubscribe$ = new Subject<void>();
+
+  constructor(
+    private _playlistService: PlaylistsService,
+    private _songService: SongsService,
+    private _router: Router,
+    private _activatedRoute: ActivatedRoute
+  ) { }
+
+  ngOnInit() {
+    this._activatedRoute.paramMap.pipe(
+      switchMap((params) => params.getAll('id'))
+    ).subscribe((data) => {
+      this._id = +data;
+    });
+
+    if (this._id) {
+      this._playlistService.getPlaylist(this._id)
+        .pipe(takeUntil(this._unsubscribe$))
+        .subscribe({
+          next: (data) => {
+            this.playlist = data;
+          },
+          error: (err) => {
+            this._router.navigateByUrl('/playlists');
+          }
+        });
+
+      this._playlistService.getPlaylistSongs(this._id)
+        .pipe(takeUntil(this._unsubscribe$))
+        .subscribe({
+          next: (data) => {
+            this.playlistSongs = data;
+          }
+        });
+    }
+    else {
+      this.playlist.id = 0;
+      this.playlist.name = 'Playlist title';
+      this.playlist.accessType = AccessType.default;
+      this.playlist.author = { id: 1 } as User; // must be current auth user here
+      this.createPlaylist();
+    }
+
+    this._searchTerms.pipe(
+      takeUntil(this._unsubscribe$),
+      debounceTime(500),
+      distinctUntilChanged(),
+      switchMap((term: string) => this._songService.getSongsByName(term))
+    ).subscribe({
+      next: (data) => {
+        this.foundSongs = data;
+      }
+    });
+  }
+
+  public ngOnDestroy() {
+    this._unsubscribe$.next();
+    this._unsubscribe$.complete();
+  }
+
+  createPlaylist() {
+    this._playlistService.createPlaylist(this.playlist)
+      .pipe(takeUntil(this._unsubscribe$))
+      .subscribe({
+        next: (data) => {
+          this.playlist = data;
+        }
+      });
+  }
+
+  editPlaylist = (editedPlaylist: EditedPlaylist) => {
+    if (editedPlaylist.name.trim() !== '') {
+      this.playlist.name = editedPlaylist.name;
+    }
+
+    this.playlist.description = editedPlaylist.description;
+    this.playlist.iconURL = editedPlaylist.iconURL;
+    this.playlist.accessType = editedPlaylist.accessType;
+
```

Comment:
Don't change each property separately 
this.playlist = { 
         desc: '',
         iconURL:  editedPlaylist.iconURL,
}